test/integration/model.js

```diff
@@ -863,17 +863,6 @@ module.exports = function(bookshelf) {
         return m.save({item: 'test'});
       });
 
-      it('does not set created_at when {method: "update"} is passed', function() {
```

Comment:
I'm not too sure about removing this test. The thing is that `created_at` mustn't change when updating a model, but your PR is producing a change right?

Reply:
Sorry for the late reply. New year, vacation etc...

Well, technically, you are right. But since Bookshelf is more Active Record rather than Data Mapper, the only way for Bookshelf to know whether the model is new or not - is to look at its ID.

If we dont have an ID attribute (its undefined, null, empty - whatever) - the model is new.
If we have an ID attribute - the model is not new.

When **new** model is saved, we perform an _**insert**_ query. When **non new** (I'll call it **dirty** from now on) model is saved, we will perform an _**update**_ query.

However the ID attribute can be set by the application. Example I've already gave: I generate the ID as UUID in my application code, rather than letting the database to generate the ID for me. This mean that **new** models can have ID, which make bookshelf _think_ that those models are **dirty** rather than **new** thus requiring bookshelf to perform an _**update**_ query when saving such model, rather than _**insert**_ query.

Assuming I bring the test back, lets look at a simple table:

```sql
CREATE TABLE IF NOT EXISTS `test` (
	`id` CHAR(36) NOT NULL,
	`created_at` datetime NOT NULL,
	`updated_at` datetime NOT NULL,
	PRIMARY KEY( `id` )
);
```

When I create a model from this table and I generate the ID (which is UUID for example), the method will always be _**update**_, because this is the way active record and bookshelf works. [Here is the code](https://github.com/skwee357/bookshelf/blob/3f7c59ec79a32de21460b03c404b90f3089c24f2/src/base/model.js#L220). And if we dont update created_at on _method=update_, the query will fail, because created_at is not nullable.

I'm open to discuss a better solution, but this is a limiting factor of active record pattern. If you determine the "new"-ness of a model by looking at the value of its ID attribute rather than doing a trip to the DB to see if record with such ID exists, you might run into this edge case. Not everyone wants to let the database manage IDs. Its not working all the time. Once again, the same example - client generated UUID.

This is a non intuitive edge case - I agree. And once again, I'm open for discussion about possible solutions, but currently this is the only solution I've found.

Reply:
I see. Very good explanation, thanks! I understand why you need to change how `created_at` is handled when updating. This could actually solve the issue of not being able to update the `created_at` column without having to resort to even more option flags, however this is a major change.

I'm currently preparing a small bug fix and documentation updates release without any major changes, but this timestamp issue will be handled on the next release after that for sure. If you don't mind I'd like to think a little bit more about this issue before merging, to avoid potentially introducing even more problems related with this functionality. Specifically, the changes in PR #900 seem to have the potential to cause conflicts.

Reply:
Awesome! Ping me if you would like to discuss or have some ideas, I'm open to listen to other solutions as well! :)